### PR TITLE
Add evaluation form to Welcome Flow

### PR DESCRIPTION
**Pull request by robertsreberski**
## Proposed changes:
* Add Evaluation step with checkbox form
* Add Evaluation "Loader" step (with timeout, currently)
* Integrate Evaluation with Welcome Banner flow

CleanShot 2024-06-19 at 15 56 06@2x
CleanShot 2024-06-19 at 15 56 18@2x

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Create a JN site with Jetpack Beta `add/welcome-banner-survey` branch selected
* In My Jetpack, notice that you have a Welcome Banner presented
* Connect the site, and notice that the next step is Evaluation form
* After submitting the form - you should see "Evaluation processing" screen for around 3 seconds (mocked operation which will be addressed in the following PR) - after that the Welcome Banner will disappear

### CODE REVIEW

File: projects/packages/my-jetpack/_inc/components/welcome-flow/EvaluationProcessingStep.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Col, Text } from '@automattic/jetpack-components';
+import { Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React from 'react';
```
Comment (CodeyGuyDylan) [inline projects/packages/my-jetpack/_inc/components/welcome-flow/EvaluationProcessingStep.tsx]: I don't think this is needed in new files 🤔

File: projects/packages/my-jetpack/_inc/components/welcome-flow/EvaluationStep.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import { Col, Button, Text } from '@automattic/jetpack-components';
+import { CheckboxControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import clsx from 'clsx';
+import React, { useCallback, useState } from 'react';
```
Comment (CodeyGuyDylan) [inline projects/packages/my-jetpack/_inc/components/welcome-flow/EvaluationStep.tsx]: I think we can avoid importing React here and in the other files

File: projects/packages/my-jetpack/_inc/components/welcome-flow/EvaluationStep.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import styles from './style.module.scss';
+
+export type EvaluationAreas = 'protect' | 'performance' | 'audience' | 'content' | 'unsure';
+
+const questions: { [ key in EvaluationAreas ]: string } = {
+	protect: __( 'Protect my site', 'jetpack-my-jetpack' ),
+	performance: __( "Improve my site's performance", 'jetpack-my-jetpack' ),
+	audience: __( 'Grow my audience', 'jetpack-my-jetpack' ),
+	content: __( 'Create quality content', 'jetpack-my-jetpack' ),
+	unsure: __( "I don't know", 'jetpack-my-jetpack' ),
```
Comment (CodeyGuyDylan) [inline projects/packages/my-jetpack/_inc/components/welcome-flow/EvaluationStep.tsx]:
```suggestion
	unsure: __( 'I don‘t know', 'jetpack-my-jetpack' ),
```

**Review by CodeyGuyDylan — APPROVED** [review]
Only very minor notes on this one from me. Everything tested well and the code looks solid

**Comment by ilonagl** [comment]
Thank you for working on this, @robertsreberski! 

Everything looks great; I’d only remove the red dot after they connect the site to Jetpack. Users did the main action we wanted them to do, and the onboarding survey is a secondary action.

File: projects/packages/my-jetpack/_inc/components/welcome-flow/index.tsx [hunk 4]
```diff
@@ -15,17 +15,37 @@ import type { FC } from 'react';
...
+
+		return 'evaluation-processing';
+	}, [ isProcessingEvaluation, siteIsRegistered ] );
 
 	const onDismissClick = useCallback( () => {
-		recordEvent( 'jetpack_myjetpack_welcome_banner_dismiss_click' );
+		recordEvent( 'jetpack_myjetpack_welcome_banner_dismiss_click', {
+			currentStep,
+			isUserConnected,
+			isSiteConnected,
```
Comment (CodeyGuyDylan) [inline projects/packages/my-jetpack/_inc/components/welcome-flow/index.tsx:44]: We can go ahead and remove this with [this PR](https://github.com/Automattic/jetpack/pull/37974) being shipped

**Review by CodeyGuyDylan — APPROVED** [review]

**Comment by CodeyGuyDylan** [comment]
@robertsreberski One comment I wanted to make before this is merged, is it weird that a user can select 1-4 specific options AND "I don't know"? I feel as though the "I don't know" option shouldn't be able to be selected if any other options are selected. I guess in that case it wouldn't be able to be in the checkbox list, maybe a separate button next to "Skip onboarding" that defaults their answer to "I don't know"? Just a thought I had

**Comment by ilonagl** [comment]
@CodeyGuyDylan shared a valid thought! @robertsreberski, how does the logic of the current recommendations work re picking 'I don't know'? I wonder if we should disable this choice if other choices are selected to prevent the use case that Dylan shared?

**Comment by robertsreberski** [comment]
> @CodeyGuyDylan shared a valid thought! @robertsreberski, how does the logic of the current recommendations work re picking 'I don't know'? I wonder if we should disable this choice if other choices are selected to prevent the use case that Dylan shared?

@ilonagl it currently adds points to promote specific products, as described in the table here: pbNhbs-aN2-p2

I was thinking that even if user selects that they want to improve the security of their website and "I don't know" option - we'll understand it as "I don't know what else I can improve on my website" and besides security products, promote also more general ones.

Because of that, currently you can select all checkboxes - I see it's worth re-considering from UX point of view. It's very easy to adjust the code to any option we'll decide to go with :)

**Pull request merged**